src/pal/src/file/directory.cpp

```diff
@@ -125,15 +125,18 @@ Return Value:
 static
 BOOL
 RemoveDirectoryHelper (
-    LPSTR lpPathName,
+    PathCharString& lpPathNamePS,
     LPDWORD dwLastError
 )
 {
     BOOL  bRet = FALSE;
     *dwLastError = 0; 
 
+    LPSTR lpPathName = lpPathNamePS.OpenStringBuffer(lpPathNamePS.GetCount(),TRUE);
     FILEDosToUnixPathA( lpPathName );
-    if ( !FILEGetFileNameFromSymLink(lpPathName))
+    lpPathNamePS.CloseBuffer(lpPathNamePS.GetCount());
+
+    if ( !FILEGetFileNameFromSymLink(lpPathNamePS))
```

Comment:
FILEGetFileNameFromSymLink might need to grow the size of lpPathNamePS, which can invalidate lpPathName. Thus lpPathName needs to be reset after calling FILEGetFileNameFromSymLink. Similar issues exist in two locations in file.cpp.